scripts/argocd-helper.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+#
+# Tests if the ArgoCD instance is available on the cluster by logging in.
+#
+# Uses the ArgoCD session, created by previsoly running "argocd login", to
+# generate a acccount token. The informaton is then stored on the ARGOCD_ENV_FILE
+# variable.
+#
+
+shopt -s inherit_errexit
+set -Eeu -o pipefail
+
+declare -r SUBCOMMAND="${1:-}"
+
+# ArgoCD hostname (FQDN) to test.
+declare -r ARGOCD_HOSTNAME="${ARGOCD_HOSTNAME:-}"
+# ArgoCD username to use for login.
+declare -r ARGOCD_USER="${ARGOCD_USER:-admin}"
+# ArgoCD password to use for login.
+declare -r ARGOCD_PASSWORD="${ARGOCD_PASSWORD:-}"
+# Environment file to store the ArgoCD crednetials.
+declare -r ARGOCD_ENV_FILE="${ARGOCD_ENV_FILE:-/rhtap/argocd/env}"
+
+fail() {
+    echo "# [ERROR] ${*}" >&2
+    exit 1
+}
+
+# Asserts the required environment variables.
+assert_variables() {
+    [[ -z "${ARGOCD_HOSTNAME}" ]] &&
+        fail "ARGOCD_HOSTNAME is not set!"
+    [[ -z "${ARGOCD_USER}" ]] &&
+        fail "ARGOCD_USER is not set!"
+    [[ -z "${ARGOCD_PASSWORD}" ]] &&
+        fail "ARGOCD_PASSWORD is not set!"
+}
+
+# Executes the ArgoCD login command.
+argocd_login() {
+    argocd login "${ARGOCD_HOSTNAME}" \
+        --grpc-web \
+        --insecure \
+        --http-retry-max="5" \
+        --username="${ARGOCD_USER}" \
+        --password="${ARGOCD_PASSWORD}"
+}
+
+# Retries a few times until the ArgoCD instance is available.
+test_argocd_login() {
+    for i in {1..30}; do
+        echo "# [${i}/30] Testing ArgoCD login on '${ARGOCD_HOSTNAME}'..."
+        argocd_login &&
+            return 0
+
+        wait=$((i * 5))
+        echo "### [${i}/30] Waiting for ${wait} seconds before retrying..."
```

Comment:
We should not sleep on the 30th loop as we won't try logging in again.

Reply:
That's a fix we need to tackle on basically all scripts, good point. Could we tackle this in a single PR, please?